client/core/core_test.go

```diff
@@ -3132,39 +3181,104 @@ func TestReadConnectMatches(t *testing.T) {
 	tracker.matches[knownID] = knownMatch
 	knownMsgMatch := &msgjson.Match{OrderID: oid[:], MatchID: knownID[:]}
 
+	// Known trade, but missing match
 	missingID := ordertest.RandomMatchID()
 	missingMatch := &matchTracker{
 		id:        missingID,
 		MetaMatch: metaMatch,
 	}
 	tracker.matches[missingID] = missingMatch
 
+	// extra match
 	extraID := ordertest.RandomMatchID()
 	extraMsgMatch := &msgjson.Match{OrderID: oid[:], MatchID: extraID[:]}
 
-	matches := []*msgjson.Match{knownMsgMatch, extraMsgMatch}
-	tracker.readConnectMatches(matches)
+	// Entirely missing order
+	loMissing, dbOrderMissing, preImgMissing, _ := makeLimitOrder(dc, true, 3*tDCR.LotSize, tBTC.RateStep*10)
+	trackerMissing := newTrackedTrade(dbOrderMissing, preImgMissing, dc, mkt.EpochLen, rig.core.lockTimeTaker, rig.core.lockTimeMaker,
+		rig.db, rig.queue, nil, nil, notify)
+	oidMissing := loMissing.ID()
+	// an active match for the missing trade
+	matchIDMissing := ordertest.RandomMatchID()
+	missingTradeMatch := &matchTracker{
+		id: matchIDMissing,
+		MetaMatch: db.MetaMatch{
+			MetaData: &db.MatchMetaData{},
+			Match:    &order.UserMatch{},
+		},
+		counterConfirms: -1,
+	}
+	trackerMissing.matches[knownID] = missingTradeMatch
+	// an inactive match for the missing trade
+	matchIDMissingInactive := ordertest.RandomMatchID()
+	missingTradeMatchInactive := &matchTracker{
+		id: matchIDMissingInactive,
+		MetaMatch: db.MetaMatch{
+			MetaData: &db.MatchMetaData{Status: order.MatchComplete},
+			Match:    &order.UserMatch{Status: order.MatchComplete},
+		},
+		counterConfirms: 1,
+	}
+	trackerMissing.matches[matchIDMissingInactive] = missingTradeMatchInactive
 
-	if knownMatch.failErr != nil {
-		t.Fatalf("error set for known and reported match")
+	srvMatches := map[order.OrderID]*serverMatches{
+		oid: {
+			tracker:    tracker,
+			msgMatches: []*msgjson.Match{knownMsgMatch, extraMsgMatch},
+		},
+		// oidMissing not included (missing!)
 	}
 
-	if missingMatch.failErr == nil {
-		t.Fatalf("error not set for missing match")
+	dc.trades = map[order.OrderID]*trackedTrade{
+		oid:        tracker,
+		oidMissing: trackerMissing,
 	}
 
-	if len(notes["order"]) != 2 {
-		t.Fatalf("expected 2 core 'order'-type notifications, got %d", len(notes["order"]))
+	exceptions := dc.compareServerMatches(srvMatches)
+	for oidExc, discrep := range exceptions {
+		t.Logf("%v: %#v\n", oidExc, discrep)
 	}
 
-	if notes["order"][0].Subject() != "Missing matches" {
-		t.Fatalf("no core notification sent for missing matches")
+	if len(exceptions) != 2 {
+		t.Fatalf("exceptions did not include both trades, just %d", len(exceptions))
 	}
 
-	if notes["order"][1].Subject() != "Match resolution error" {
-		t.Fatalf("no core notification sent for unknown matches")
+	if exc, ok := exceptions[oid]; !ok {
+		t.Fatalf("exceptions did not include trade %v", oid)
+	} else {
```

Comment:
No need for the `else` block.

Reply:
Strange that I did it this way.  Had to be else because `exc` scope, but I'll pull the map access out of the if-else scope